plugins/modules/win_acl_inheritance.ps1

```diff
@@ -1,75 +1,156 @@
+    param (
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [System.Security.AccessControl.NativeObjectSecurity] $acl
+    )
 
-                ForEach ($explicitRrule in $objACL.Access) {
-                    If ($explicitRrule.IsInherited) {
-                        Continue
-                    }
+    Begin {}
 
-                    If (
-                        ($inheritedRule.FileSystemRights -eq $explicitRrule.FileSystemRights) -And
-                        ($inheritedRule.AccessControlType -eq $explicitRrule.AccessControlType) -And
-                        ($inheritedRule.IdentityReference -eq $explicitRrule.IdentityReference) -And
-                        ($inheritedRule.InheritanceFlags -eq $explicitRrule.InheritanceFlags) -And
-                        ($inheritedRule.PropagationFlags -eq $explicitRrule.PropagationFlags)
-                    ) {
-                        $objACL.RemoveAccessRule($explicitRrule)
+    Process {
+        $properties = $acl.Access |
+            Get-Member -MemberType Property |
+            Where-Object { $_.Name -ne 'IsInherited' } |
+            Select-Object -ExpandProperty Name
+
+        ForEach ($inheritedRule in $($acl.Access | Where-Object { $_.IsInherited })) {
+            ForEach ($explicitRule in $($acl.Access | Where-Object { -not $_.IsInherited })) {
+                If ($null -eq (Compare-Object -ReferenceObject $explicitRule -DifferenceObject $inheritedRule -Property $properties)) {
+                    if ($PSCmdlet.ShouldProcess($acl)) {
+                        $acl.RemoveAccessRule($explicitRule)
+                    }
+                    else {
+                        Write-Verbose "WhatIf: Performing remove explicit duplicate rule: $explicitRule"
                     }
                 }
             }
         }
+        return $acl
+    }
+
+    End {}
+}
 
-        Set-ACL -LiteralPath $path -AclObject $objACL -WhatIf:$check_mode
-        $result.changed = $true
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$path = $module.Params.path
+$reorganize = $module.Params.reorganize
+$state = $module.Params.state
+$check_mode = $module.CheckMode
+
+$module.Result.changed = $false
+
+$pathQualifier = Split-Path -Path $path -Qualifier -ErrorAction SilentlyContinue
+
+if ($pathQualifier -eq 'HKCR:' -and (-not (Test-Path -LiteralPath HKCR:\))) {
+    $null = New-PSDrive -Name HKCR -PSProvider Registry -Root HKEY_CLASSES_ROOT
+}
+if ($pathQualifier -eq 'HKU:' -and (-not (Test-Path -LiteralPath HKU:\))) {
+    $null = New-PSDrive -Name HKU -PSProvider Registry -Root HKEY_USERS
+}
+if ($pathQualifier -eq 'HKCC:' -and (-not (Test-Path -LiteralPath HKCC:\))) {
+    $null = New-PSDrive -Name HKCC -PSProvider Registry -Root HKEY_CURRENT_CONFIG
+}
+
+If (-Not (Test-Path -LiteralPath $path)) {
+    $module.FailJson("$path does not exist")
+}
+
+try {
+    $acl = Get-AclEx -LiteralPath $path
+}
+catch {
+    $module.FailJson($_.ToString())
```

Comment:
It's best to provide a message saying what this was trying to do with the exception message at the end. The `FailJson` also accepts an optional 2nd argument which is the Exception or ErrorRecord which is used to display the full error + stacktrace when running with a higher verbosity.

```suggestion
    $module.FailJson("Failed to find '$path': $($_.ToString())", $_)
```